torch/utils/cpp_extension.py

```diff
@@ -289,13 +338,20 @@ def unix_wrap_single_compile(obj, src, ext, cc_args, extra_postargs, pp_opts):
             try:
                 original_compiler = self.compiler.compiler_so
                 if _is_cuda_file(src):
-                    nvcc = _join_cuda_home('bin', 'nvcc')
+                    nvcc = (_join_rocm_home('bin', 'hipcc') if is_hip_extension else _join_cuda_home('bin', 'nvcc'))
                     if not isinstance(nvcc, list):
                         nvcc = [nvcc]
                     self.compiler.set_executable('compiler_so', nvcc)
                     if isinstance(cflags, dict):
                         cflags = cflags['nvcc']
-                    cflags = unix_cuda_flags(cflags)
+                    if is_hip_extension:
+                        cflags = COMMON_HIPCC_FLAGS + cflags + _get_rocm_arch_flags(cflags)
+                    else:
+                        cflags = unix_cuda_flags(cflags)
+                elif is_hip_extension:
```

Comment:
nit: I think this is slightly easier to read if we have the following:
```
if _is_cuda_file(src):
  ...
elif isinstance(flags,dict):
  cflags = cflags['cxx']
if is_hip_extension:
   cflags = COMMON_HIPCC_FLAGS + cflags
```
so that we don't have to duplicate the `isinstance(cflags, dict)` check logic

Reply:
Simplified the condition